app/fileUploadParsers/LifetimeAllowanceParser.scala

```diff
@@ -17,26 +17,74 @@
 package fileUploadParsers
 
 import com.google.inject.Inject
+import config.FrontendAppConfig
 import forms.MemberDetailsFormProvider
-import models.UserAnswers
-import pages.chargeD.MemberDetailsPage
+import forms.chargeD.ChargeDetailsFormProvider
+import models.chargeD.ChargeDDetails
+import models.{MemberDetails, Quarters}
+import pages.chargeD.{ChargeDetailsPage, MemberDetailsPage}
+import play.api.data.Form
+import play.api.i18n.Messages
+import play.api.libs.json.Json
+
+import java.time.LocalDate
 
 class LifetimeAllowanceParser @Inject()(
-                                       memberDetailsFormProvider: MemberDetailsFormProvider
-                                     ) extends Parser {
+                                         memberDetailsFormProvider: MemberDetailsFormProvider,
+                                         chargeDetailsFormProvider: ChargeDetailsFormProvider,
+                                         config: FrontendAppConfig
+                                       ) extends Parser {
+  //scalastyle:off magic.number
+  override protected def validHeader: String = config.validLifeTimeAllowanceHeader
 
-  override protected val totalFields:Int = 7
+  override protected val totalFields: Int = 6
 
-  override protected def validateFields(ua:UserAnswers, index: Int, chargeFields: Array[String]) : Either[ParserValidationErrors, UserAnswers] = {
-    val m = Map(
-      "firstName" -> firstNameField(chargeFields),
-      "lastName" -> lastNameField(chargeFields),
-      "nino" -> ninoField(chargeFields)
-    )
-    val form = memberDetailsFormProvider.apply()
-    form.bind(m).fold(
-      formWithErrors => Left(ParserValidationErrors(index, formWithErrors.errors.map(_.message))),
-      value => Right(ua.setOrException(MemberDetailsPage(index), value))
+  private final object ChargeDetailsFieldNames {
+    val dateOfEventDay: String = "dateOfEvent.day"
+    val dateOfEventMonth: String = "dateOfEvent.month"
+    val dateOfEventYear: String = "dateOfEvent.year"
+    val taxAt25Percent: String = "taxAt25Percent"
+    val taxAt55Percent: String = "taxAt55Percent"
+    val dateOfEvent: String = "dateOfEvent"
+  }
+
+  private def chargeDetailsValidation(startDate: LocalDate,
+                                      index: Int,
+                                      chargeFields: Array[String])(implicit messages: Messages): Either[Seq[ParserValidationError], ChargeDDetails] = {
+
+    splitDayMonthYear(chargeFields(3)) match {
```

Comment:
on line 55, you could assign year, month and day to a variables instead of using pattern match. I think pattern match would be used if you have multiple matches, but in this case you have a single match statement. 
I suggest you use example below.
val (year, month, day) = splitDayMonthYear

Reply:
Good point. I have changed it to work as per your example.